### PR TITLE
Initial work on new video dock design

### LINKED ISSUE #5056: Inconsistency in webcam share button.
button-video
button-video-off

### PR DESCRIPTION
**Pull request by gutobenn**
This is the initial work on the HTML5 video dock design (#4934). It also fixes #5056.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/media/styles.scss [hunk 1]
```diff
@@ -29,16 +29,37 @@
...
     flex-basis: 15%;
+
   }
 }
 
+.moreThan4Videos{
+  top: 63px;
+  right: 0;
+  position: fixed;
+  height: calc(100% - 143px);
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/media/styles.scss]: Magic numbers. You probably want to use percentage/rem for everything.

File: bigbluebutton-html5/client/compatibility/adjust-videos.js [hunk 2]
```diff
@@ -55,30 +55,51 @@
...
 
-    const videos = $("#" + tagId + " video:visible");
+    const videos = $("#" + tagId + " > div:visible");
+    const isPortrait = ( $(document).width() < $(document).height() );
+
+    if (isPortrait) {
+      // If currently displaying a presentation
+      if ( $("#" + presentationAreaDataId).length ) {
+        e.css({
+          "margin-top": $('#' + presentationAreaDataId).offset().top - 221,
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/client/compatibility/adjust-videos.js]: What is 221?

File: bigbluebutton-html5/client/compatibility/adjust-videos.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 
 (function() {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/client/compatibility/adjust-videos.js]: Just a heads up that this is changing the `style` and `class` of some elements. This is not a good idea when using React. We probably need to move all this logic to some sort of layout component and rely less on javascript calculations using flexbox.

Comment (oswaldoacauan) [reply bigbluebutton-html5/client/compatibility/adjust-videos.js]: And is using jQuery :puke:

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 4]
```diff
@@ -40,19 +41,39 @@ class VideoElement extends Component {
...
+    }
     return (
-      <video
-        id={`video-elem-${this.props.videoId}`}
-        width={320}
-        height={240}
-        autoPlay="autoPlay"
-        playsInline="playsInline"
-      />
+      <div className={styles.videoContainer + " " + cssClass} >
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: Please make use of the [`classnames`](https://github.com/JedWatson/classnames) package.

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 5]
```diff
@@ -40,19 +41,39 @@ class VideoElement extends Component {
   }
 
   render() {
+    let cssClass;
+    if (this.props.shared || !this.props.localCamera) {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: Always try to [destructure assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) when using properties multiple times.

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 6]
```diff
@@ -40,19 +41,39 @@ class VideoElement extends Component {
...
-        height={240}
-        autoPlay="autoPlay"
-        playsInline="playsInline"
-      />
+      <div className={styles.videoContainer + " " + cssClass} >
+        { this.props.localCamera ?
+            <video id="shareWebcam" muted={true} autoPlay={true} playsInline={true} />
+          :
+            <video id={`video-elem-${this.props.videoId}`} autoPlay={true} playsInline={true} />
+        }
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]:
```
const videoElementId = localCamera ? 'shareWebcam' : `video-elem-${videoId}`'
<video id={videoElementId} muted={localCamera} autoPlay playsInline />
```

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 7]
```diff
@@ -40,19 +41,39 @@ class VideoElement extends Component {
...
+            className={styles.pauseButton}
+            icon={'unmute'}
+            size={'sm'}
+            circle
+            onClick={() => {}}
+          />*/}
+        </div>
+      </div>
     );
   }
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: Please move this component into another file. Try to keep each file with a single component.

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 8]
```diff
@@ -318,7 +350,7 @@ class VideoDock extends Component {
 
     let peerObj;
     if (shareWebcam) {
-      options.localVideo = this.refs.videoInput;
+      options.localVideo = document.getElementById("shareWebcam");
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: You probably should keep using refs.

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 9]
```diff
@@ -419,9 +451,11 @@ class VideoDock extends Component {
 
   destroyVideoTag(id) {
     const videos = this.state.videos;
+    const userNames = this.state.userNames;
 
     delete videos[id];
-    this.setState({ videos });
+    delete userNames[id]
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: This is wrong. By using `delete` you are actually mutating the current state in memory.
Please do something like this:
```
this.setState({
  userNames: userNames.filter((_,i) => i !== id),
})
```

PS. Fix it for the `videos` too.

File: bigbluebutton-html5/imports/ui/components/video-dock/styles.scss [hunk 10]
```diff
@@ -2,21 +2,87 @@
...
-  left: 0;
-
   background-size: cover;
   background-position: center;
-  box-shadow: 0 0 5px rgba(0, 0, 0, .5);
-  border-radius: .2rem;
+  text-align: center;
+  
+  @include mq($portrait) {
+    height: 96px;
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/styles.scss]: magic number again.

File: bigbluebutton-html5/imports/ui/components/video-dock/styles.scss [hunk 11]
```diff
@@ -2,21 +2,87 @@
...
+
+.pauseButton {
+  float: right;
+
+  span:last-child {
+    display: none;
+  } // TODO fix it better
+
+  span:first-child {
+    background: #D8D8D8;
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/video-dock/styles.scss]: Please use a variable from the globals.

**Comment by antobinary** [comment]
Some of the requested changes were applied by @oswaldoacauan in #5141 
@gutobenn please apply the remaining requested changes through a new PR

**Comment by gutobenn** [comment]
Ok
Thanks for the review, @oswaldoacauan! 👍

**Pull request merged**